### PR TITLE
add automated reward admin functions and panel

### PR DESCRIPTION
**Pull request by DrillableBit**
---
<a href="https://app.blacksmith.sh/aiarena/codesmith/aiarena-web/pr/1137"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1788633172&installation_model_id=430647&pr_number=1137&repository=aiarena%2Faiarena-web&return_to=https%3A%2F%2Fgithub.com%2Faiarena%2Faiarena-web%2Fpull%2F1137&signature=66008987b13c719d1fca34e612fd3350c9464490c904eaa315136615c39f7a5a"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

File: aiarena/core/models/award_set.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import logging
+
+from django.db import models
+
+
+logger = logging.getLogger(__name__)
+
+
+class AwardSet(models.Model):
```
Comment (lladdy) [inline aiarena/core/models/award_set.py:11]: Now that we have more people contributing to the website, lets start adding docstrings describing what the models represent.

File: aiarena/core/models/award_set_item.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+from django.db import models
+
+from .award_set import AwardSet
+from .trophy_icon import TrophyIcon
+from .trophy import TROPHY_CONDITIONS
+
+logger = logging.getLogger(__name__)
+
+
+class AwardSetItem(models.Model):
```
Comment (lladdy) [inline aiarena/core/models/award_set_item.py:14]: Same as previous comment - add docstring.

File: aiarena/core/models/trophy.py [hunk 3]
```diff
@@ -4,12 +4,39 @@
...
+    ("top_5", "Top 5"),
+    ("top_10", "Top 10"),
+    ("top_15", "Top 15"),
+    ("top_20", "Top 20"),
+    ("participant", "Participant"),
+    ("custom", "Custom"),
+)
+
+
 class Trophy(models.Model):
```
Comment (lladdy) [inline aiarena/core/models/trophy.py:25]: Rather than me taking ages to ask a bunch of questions, lets do a call around the data modelling when you're free.

**Review by lladdy — CHANGES_REQUESTED** [review]
We need tests! At minimum for the backend, but ideally for the frontend as well.

**Comment by lladdy** [comment]
Suggestions
- Deploy, run and validate update_legacy_trophies - so that this simple step is done.
- On trophy model, instead of having the condition field, we can link back to the AwardSetItem that created it.
- Move conditions to live with award sets
- Tests
